### PR TITLE
Path Planning Fixes

### PR DESCRIPTION
**Pull request by barulicm**
This branch includes the changes in #206. That PR should be merged first.

Building off of the work @fourpenny did to add testing for the path planner, I've spent some time fixing bugs and adding some path post-processing to make sure our paths are less likely to have any weirdness in them. My main focus was getting rid of loops (where the path crosses over itself) and switchbacks (where the path doubles back in the direction it came from).

This PR:
- Adds unit tests for the path planner
- Adds a test executable that runs randomized path planning scenarios with a ball and one opponent robot until a problematic path is found
- Adds post processing steps to the path planner to remove loops and shortcuts between path points when possible
- Adds a flag to tell the path planner whether it should ignore obstacles that collide with the start position, which might happen when you're trying to escape a keep out zone or touching another robot.
- Changes the timeout behavior so the returned path stops where the first collision occurs. This allows the robot to start moving without following a path that would cause a collision. Footprint inflation should prevent the collision point from actually causing a collision.

### CODE REVIEW

**Review by fourpenny — APPROVED** [review]
I'm fine with this, even if you don't change anything, since pretty much anything I said was style/documentation related

File: ateam_kenobi/test/path_planner_test.cpp [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+  planner_options.use_default_obstacles = false;
+  expected_path = {start, goal};
+  runTest();
+}
+
+TEST_F(GetPathTest, PlanAroundOneObstacle) {
+  start = ateam_geometry::Point(0, 0);
+  goal = ateam_geometry::Point(2, 2);
+  obstacles.push_back(ateam_geometry::makeCircle(ateam_geometry::Point(1, 1), 0.1));
+  expected_path = {start, ateam_geometry::Point(1.23762, 0.774008), goal};
```
[the remark this answers is not in the document]
Comment (barulicm) [reply ateam_kenobi/test/path_planner_test.cpp:115]: yes, this path planning approach is completely deterministic

File: ateam_kenobi/test/path_planner_test.cpp [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+}
+
+TEST_F(GetPathTest, DisallowGoalsInOpponentDefenseArea)
+{
+  start = ateam_geometry::Point(0, 0);
+  goal = ateam_geometry::Point(4, 0);
+  runTest();
+}
+
+TEST_F(GetPathTest, AllowIgnoringStartObstacles)
```
[the remark this answers is not in the document]
Comment (barulicm) [reply ateam_kenobi/test/path_planner_test.cpp:162]: good idea. Done.

File: ateam_kenobi/src/path_planning/path_planner.cpp [hunk 3]
```diff
@@ -227,4 +265,91 @@ void PathPlanner::addDefaultObstacles(
...
+      ind2 = ind1 + 2;
+      if (path.size() < 4) {
+        // path now too small to have loops
+        return;
+      }
+    }
+  }
+}
+
+void PathPlanner::trimPathAfterCollision(
```
[the remark this answers is not in the document]
Comment (barulicm) [reply ateam_kenobi/src/path_planning/path_planner.cpp:338]: Hmmm not sure the best way to handle this. The fact that planning timed out isn't really part of this function. If I was trying to figure out when it's used, I'd just use "find all references" in IDE of choice and see that it's called in the timeout branch.

**Pull request merged**